pygmt/conftest.py

```diff
@@ -0,0 +1,11 @@
+"""
+conftest.py for pytest.
+"""
+
+import numpy as np
+from packaging.version import Version
+
+# Keep this until we require numpy to be >=2.0
+# Address https://github.com/GenericMappingTools/pygmt/issues/2628.
+if Version(np.__version__) >= Version("2.0.0.dev0+git20230726"):
+    np.set_printoptions(legacy="1.25")
```

Comment:
https://github.com/numpy/numpy/blob/1d49c7f7ff527c696fc26ab2278ad51632a66660/numpy/_core/arrayprint.py#L91
`legacy` can be `1.25` but it's not well documented at https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html#numpy.set_printoptions, so we need to ignore the type error:

```suggestion
    np.set_printoptions(legacy="1.25")  # type: ignore[arg-type]
```